pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/observables/ObservablesTNCuda.hpp

```diff
@@ -0,0 +1,541 @@
+            numStateModesLocal.insert(numStateModesLocal.end(),
+                                      ob->getNumStateModes().front().begin(),
+                                      ob->getNumStateModes().front().end());
+
+            stateModesLocal.insert(stateModesLocal.end(),
+                                   ob->getStateModes().front().begin(),
+                                   ob->getStateModes().front().end());
+
+            dataLocal.insert(dataLocal.end(), ob->getMetaData().front().begin(),
+                             ob->getMetaData().front().end());
+        }
+
+        BaseType::numStateModes_.emplace_back(numStateModesLocal);
+        BaseType::stateModes_.emplace_back(stateModesLocal);
+        BaseType::metaData_.emplace_back(dataLocal);
+
+        std::unordered_set<std::size_t> wires;
+        for (const auto &ob : obs_) {
+            const auto ob_wires = ob->getWires();
+            for (const auto wire : ob_wires) {
+                PL_ABORT_IF(wires.contains(wire),
+                            "All wires in observables must be disjoint.");
+                wires.insert(wire);
+            }
+        }
+        all_wires_ = std::vector<std::size_t>(wires.begin(), wires.end());
+        std::sort(all_wires_.begin(), all_wires_.end());
+    }
+
+    /**
+     * @brief Convenient wrapper for the constructor as the constructor does not
+     * convert the std::shared_ptr with a derived class correctly.
+     *
+     * This function is useful as std::make_shared does not handle
+     * brace-enclosed initializer list correctly.
+     *
+     * @param obs List of observables
+     * @return std::shared_ptr<TensorProdObsTNCuda<StateTensorT>>
+     */
+    static auto create(
+        std::initializer_list<std::shared_ptr<ObservableTNCuda<StateTensorT>>>
+            obs) -> std::shared_ptr<TensorProdObsTNCuda<StateTensorT>> {
+        return std::shared_ptr<TensorProdObsTNCuda<StateTensorT>>{
+            new TensorProdObsTNCuda(std::move(obs))};
+    }
+
+    /**
+     * @brief Convenient wrapper for the constructor as the constructor does not
+     * convert the std::shared_ptr with a derived class correctly.
+     *
+     * This function is useful as std::make_shared does not handle
+     * brace-enclosed initializer list correctly.
+     *
+     * @param obs List of observables
+     * @return std::shared_ptr<TensorProdObsTNCuda<StateTensorT>>
+     */
+    static auto
+    create(std::vector<std::shared_ptr<ObservableTNCuda<StateTensorT>>> obs)
+        -> std::shared_ptr<TensorProdObsTNCuda<StateTensorT>> {
+        return std::shared_ptr<TensorProdObsTNCuda<StateTensorT>>{
+            new TensorProdObsTNCuda(std::move(obs))};
+    }
+
+    /**
+     * @brief Get the number of operations in observable.
+     *
+     * @return std::size_t
+     */
+    [[nodiscard]] auto getSize() const -> std::size_t { return obs_.size(); }
+
+    /**
+     * @brief Get the wires for each observable operation.
```

Comment:
```suggestion
     * @brief Get the wires of each term in the observable.
```

Reply:
Since it tensorprod obs, let's keep it as it is, which is aligned with obs docstring in obs base class in sv backends.